tests/ga/test_exthandlers.py

```diff
@@ -72,3 +72,44 @@ def test_parse_extension_status01(self):
         self.assertEqual('error', ext_status.status)
         self.assertEqual(0, ext_status.sequenceNumber)
         self.assertEqual(0, len(ext_status.substatusList))
+
+    @patch('azurelinuxagent.ga.exthandlers.ExtHandlerInstance.get_largest_seq_no')
+    def assert_extension_sequence_number(self,
+                                         patch_get_largest_seq,
+                                         goal_state_sequence_number,
+                                         disk_sequence_number,
+                                         expected_sequence_number):
+        ext = Extension()
+        ext.sequenceNumber = goal_state_sequence_number
+        patch_get_largest_seq.return_value = disk_sequence_number
+
+        ext_handler_props = ExtHandlerProperties()
+        ext_handler_props.version = "1.2.3"
+        ext_handler = ExtHandler(name='foo')
+        ext_handler.properties = ext_handler_props
+
+        instance = ExtHandlerInstance(ext_handler=ext_handler, protocol=None)
+        seq, path = instance.get_status_file_path(ext)
+
+        self.assertEqual(expected_sequence_number, seq)
+        if seq > -1:
+            self.assertTrue(path.endswith('/foo-1.2.3/status/{0}.status'.format(expected_sequence_number)))
+        else:
+            self.assertIsNone(path)
+
+    def test_extension_sequence_number(self):
+        self.assert_extension_sequence_number(goal_state_sequence_number="12",
+                                              disk_sequence_number=366,
```

Comment:
I understand the value in asserting what the agent will do in this case, but I don't believe the answer is *correct*. Extensions do not know what settings to execute when enabled. Extensions only know to execute the latest, and the latest is arbitrary. Extensions execute a directory listing, get the number of the file, sort, and take the last one in ascending order. The agent never explicitly says which settings to execute, which is a mistake, but that ship has sailed.

I would keep the test as is because I believe the only way to get into this situation is if CRP's sequence for an extension install stops being monotonically increasing.  If CRP is wrong then we have other issues. It is worth adding a telemetry event for the case where get_largest_seq_no() > extension.sequenceNumber as a signal that things have gone pear shaped.

Reply:
Added telemetry